drivers/auxiliary/ipx800.h

```diff
@@ -0,0 +1,224 @@
+This file is part of the IPX800 INDI Driver.
+A driver for the IPX800 (GCE Electronics - https://www.gce-electronics.com)
+
+Copyright (C) 2024 Arnaud Dupont (aknotwot@protonmail.com)
+
+IPX800 INDI Driver is free software : you can redistribute it
+and / or modify it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, either version 3 of the License,
+or (at your option) any later version.
+
+IPX800 INDI Driver is distributed in the hope that it will be
+useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with IPX800 INDI Driver.  If not, see
+< http : //www.gnu.org/licenses/>.
+
+*******************************************************************************/
+#pragma once
+
+#include "defaultdevice.h"
+#include <indioutputinterface.h>
+#include <indiinputinterface.h>
+#include <indidevapi.h>
+#include <indiapi.h>
+ 
+class Ipx800 : public INDI::DefaultDevice, public INDI::InputInterface, public INDI::OutputInterface
+ 
+{
+  public:
+  
+	Ipx800();
+    virtual ~Ipx800() override = default;
+	
+	virtual bool initProperties() override;
+	virtual bool updateProperties() override;
+	const char *getDefaultName() override;
+	
+	virtual bool ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n) override;
+    virtual bool ISNewText(const char *dev, const char *name, char *texts[], char *names[], int n) override;
+    virtual bool ISNewNumber(const char *dev,const char *name,double values[],char *names[],int n) override;
+	   
+	//TO CONFIRM
+	
+    virtual bool ISSnoopDevice(XMLEle *root) override;
+    virtual void ISGetProperties(const char *dev) ;
+ protected:
+	
+	bool Handshake();
+	bool Connect() override;
+    bool Disconnect() override;
+	void TimerHit() override;
+	
+    enum IPX800_command {
+       GetR   = 1 << 0,
+       GetD  = 1 << 1,
+       SetR = 1 << 2,
+       ClearR = 1 << 3
+   } ;
+       
+	///////////////////////////////////////////
+	// IPX800 Communication
+	///////////////////////////////////////////
+	bool updateIPXData();
+    void updateObsStatus();
+    bool readCommand(IPX800_command);
+    bool writeCommand(IPX800_command, int toSet);
+    bool checkAnswer();
+    void readAnswer();
+    void recordData(IPX800_command command);
+    bool writeTCP(std::string toSend);
+	bool firstFonctionTabInit();
+	
+    virtual bool UpdateDigitalInputs() override;
+    virtual bool UpdateAnalogInputs() override; // IPX800 Analog Inputs not managed
+    virtual bool UpdateDigitalOutputs() override;
+    virtual bool CommandOutput(uint32_t index, OutputState command) override;
+	virtual bool saveConfigItems(FILE *fp) override;
+	
+  private:
+	bool roofPowerManagement = false;
```

Comment:
Not needed

Reply:
useful for additionnal function (engine power management)